### DISCUSSION TITLE
Contribution graph missing work

### DISCUSSION THREAD

**Discussion by TootEgozy** [community/community#65760] [Other Feature Feedback, Questions, & Ideas] [answered]
### Contribution graph

Question

### Contribution graph not showing deleted branches

Hi, I've noticed that deleting branches results in my contribution graph on my page turning into blank squares, even though it previously displayed my commits.

For instance, this week, I pushed new commits to my project daily, except for Friday. I branch off from my development branch, work on the feature, merge it into the development branch, and then delete the feature branch. As far as I know, this is considered good practice. However, it causes my contribution graph to go blank on the days I worked on the deleted branches.

Is there a different issue at play? If not, must I retain all my merged branches rather than deleting them? It's important to me that my contribution graph reflects my consistent work.

Thank you.

**Comment by Binary-Ninja-007** [comment]
Hey there! 😊

I understand that you're concerned about your contribution graph not showing the work you've done on deleted branches. Let me explain why this happens and how you can handle it.

When you delete a branch in Git, the associated commits are no longer directly linked to any branch, and GitHub's contribution graph only counts commits that are part of a branch. That's why the squares on your graph become blank for the days you worked on the deleted branches.

However, there's no need to worry! There are a couple of options you can consider to maintain a consistent contribution graph:

Don't delete branches: As you mentioned, you can choose not to delete your branches after merging them. By keeping the branches, their commits will still be associated with them, and your contribution graph will continue to reflect your work accurately. This is a common practice for those who want to maintain a detailed commit history.

Create long-lived branches: Instead of creating and deleting branches for each feature, you can create long-lived branches that serve as containers for multiple features or tasks. This way, you can continue to merge your work into the development branch while keeping the long-lived branches intact.

Rebase and force push: If you want to maintain a clean branch structure while also having your contributions reflected in the graph, you can consider rebasing your feature branches onto the latest state of the development branch. This way, the commits from the deleted branches will be reapplied on top of the main branch, and your contribution graph will display the activity accurately.

Remember, the contribution graph is just a visual representation of your activity, and the actual commits are still present in the repository's history, regardless of what's shown on the graph. So, feel free to choose the approach that best fits your workflow and preferences.

I hope this helps! If you have any more questions, feel free to ask. 😉

**Comment by TootEgozy** [comment]
Thank you for your detailed answer :)
